File: 2516-take-k-of-each-character-from-left-and-right/2516-take-k-of-each-character-from-left-and-right.py
class Solution:
    def takeCharacters(self, s: str, k: int) -> int:
        n = len(s)
        vis = [0]*3
        
        for x in s:
            vis[ord(x)-ord('a')]+=1
        if min(vis) < k:
            return -1
        
        l,r,ans =0,0,0
      
        
        while r < n:
            vis[ord(s[r])-ord('a')]-=1
            while min(vis) < k:
                vis[ord(s[l])-ord('a')]+=1
                l+=1
            ans = max(ans,r-l+1)
            r+=1
        return n - ans;

File: 2516-take-k-of-each-character-from-left-and-right/2516-take-k-of-each-character-from-left-and-right.cpp
class Solution {
public:
    int takeCharacters(string s, int k) {
        int n = s.size();
        vector<int>vis(3);
        for(auto it : s)
            vis[it-'a']++;
        if(*min_element(vis.begin(),vis.end())<k)return -1;
        // vis[0] vis[1] vis[2]
        int l = 0,r = 0,ans = 0;
        while(r < n){
            vis[s[r]-'a']--;
            while(*min_element(vis.begin(),vis.end())<k){
                vis[s[l]-'a']++;
                l++;
            }
            ans = max(ans,r-l+1);
            r++;
        }
        return n - ans;
    }
};